.github/workflows/llvm-quick-fuzz.yml

```diff
@@ -11,40 +11,55 @@ jobs:
+              , "https://releases.llvm.org/6.0.0/clang+llvm-6.0.0-x86_64-apple-darwin.tar.xz"
+              , "https://releases.llvm.org/5.0.2/clang+llvm-5.0.2-x86_64-apple-darwin.tar.xz"
+              , "https://releases.llvm.org/4.0.1/clang+llvm-4.0.1-x86_64-apple-darwin.tar.xz"
+              , "https://releases.llvm.org/3.9.0/clang+llvm-3.9.0-x86_64-apple-darwin.tar.xz"
+              , "https://releases.llvm.org/3.8.0/clang+llvm-3.8.0-x86_64-apple-darwin.tar.xz"
+              , "https://releases.llvm.org/3.7.0/clang+llvm-3.7.0-x86_64-apple-darwin.tar.xz"
+              , "https://releases.llvm.org/3.6.2/clang+llvm-3.6.2-x86_64-apple-darwin.tar.xz"
+              , "https://releases.llvm.org/3.5.2/clang+llvm-3.5.2-x86_64-apple-darwin.tar.xz"
               ]
-    name: llvm-quick-fuzz - GHC v${{ matrix.ghc }} - ${{ matrix.os }}
+    name: llvm-quick-fuzz - ${{ matrix.llvm }}
     steps:
       - uses: actions/checkout@v2
-        with:
-          submodules: true
 
       - uses: actions/setup-haskell@v1
         id: setup-haskell
         with:
           ghc-version: ${{ matrix.ghc }}
 
-      - uses: actions/cache@v2
-        name: Cache cabal store
-        with:
-          path: |
-            ${{ steps.setup-haskell.outputs.cabal-store }}
-            dist-newstyle
-          key: cabal-${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles(format('cabal.GHC-{0}.config', matrix.ghc)) }}-${{ github.sha }}
-          restore-keys: |
-            cabal-${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles(format('cabal.GHC-{0}.config', matrix.ghc)) }}-
-
       - shell: bash
         run: brew install csmith creduce
 
-      - shell: bash
-        run: curl ${{ matrix.llvm }} | tar xf
+      - uses: actions/checkout@v2
+        with:
+          repository: elliottt/llvm-pretty
+          path: llvm-pretty
 
       - shell: bash
-        run: mv clang+llvm-*-x86_64-apple-darwin llvm && echo "$PWD/llvm/bin" >> $GITHUB_PATH
+        name: Install LLVM
+        run: |
+          curl -L ${{ matrix.llvm }} -o llvm.tar.xz
+          tar xzf llvm.tar.xz
+          mv clang+llvm-* llvm
+          echo "$PWD/llvm/bin" >> $GITHUB_PATH
 
       - shell: bash
+        name: Build
         run: cabal build -ffuzz llvm-disasm fuzz-llvm-disasm
 
-#      - shell: bash
-#        run: cabal run fuzz-llvm-disasm -- -n 5 --junit-xml results.xml -o results --collapse --clang-flags="-O -w" --clang-flags="-O -w -g" -c "$PWD/llvm/bin/clang"
+      - shell: bash
+        run: ln -s $PWD/llvm/bin/clang $PWD/llvm/bin/clangy
+
+      - shell: bash
+        name: Run fuzzing
+        run: |
+          cabal exec -ffuzz fuzz-llvm-disasm -- -n 5 --junit-xml results.xml -o results --collapse -I /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include --clang-flags="-O -w" --clang-flags="-O -w -g" -c clangy
+          test $(grep "failures=\"0\"" results.xml | wc -l) -eq 2
+        env:
+          CSMITH_PATH: "/usr/local/Cellar/csmith/2.3.0/include/csmith-2.3.0"
```

Comment:
Is there any way to automate this in case the `brew install csmith` gets a different version or the location changes?

Maybe something like `CSMITH_PATH: "ls $(type -p csmith)/../include/* | tr '\n' ':'` to check relative to the binary (untested, I don't know what the actual would be)?

Reply:
Yeah, I was wondering about this, too. I don't like it being so fragile. I'll see what I can do to make it better.